src/view/MainViewManager.js

```diff
@@ -846,6 +847,62 @@ define(function (require, exports, module) {
     }
 
     /**
+     * swaps current views to opposite panes
+     */
+    function swapPaneContent() {
+
+        var activeFileInactiveView,
+            inactiveFileActiveView,
+            activePaneId = getActivePaneId(),
+            inactivePaneId = activePaneId === FIRST_PANE ? SECOND_PANE : FIRST_PANE,
+            activePane = _getPane(activePaneId),
+            inactivePane = _getPane(inactivePaneId),
+            activeFile = activePane.getCurrentlyViewedFile(),
+            inactiveFile = inactivePane.getCurrentlyViewedFile();
+
+        // Check if one of the panes is empty move and open currently viewed  file in opposite pane.
+        // if both are empty do nothing.
+        if (!activeFile || !inactiveFile) {
+            if (!activeFile && inactiveFile) {
+                _moveView(inactivePaneId, activePaneId, inactiveFile, 0);
+            } else if (!inactiveFile && activeFile) {
+                _moveView(activePaneId, inactivePaneId, activeFile, 0);
+            }
+
+        } else {
+
+            // check if currently viewed file is present in opposing pane.  If it is open the file.
+            // if it is not add it to the opposite pane and open.
+            activeFileInactiveView = inactivePane.getViewForPath(activeFile.fullPath);
+            inactiveFileActiveView = activePane.getViewForPath(inactiveFile.fullPath);
+
+            if (inactiveFileActiveView) {
```

Comment:
Yeah, my idea was something in line with the rewrite but just calling the method with `pane, file` arguments instead of adding it to the prototype. 

The original code is fine by me too if you think so too, you call @SteveMieskoski 👍